inbm-lib/inbm_common_lib/utility.py

```diff
@@ -205,7 +205,9 @@ def is_within_directory(directory: str, target: str) -> bool:
     return prefix == abs_directory
 
 
-def safe_extract(tarball: tarfile.TarFile, path: str = ".", members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
+def safe_extract(tarball: tarfile.TarFile,
+                 path: str = ".",
+                 members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
     """Avoid path traversal when extracting tarball
```

Comment:
**Best Practice**
**Suggestion:** Use a context manager when working with files to ensure that the file is properly closed after its suite finishes executing.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
with tarfile.open(tarball) as tar:
    logger.debug(f"Extract {tarball}.")
    safe_extract(tar, path=TEMP_EXT_FOLDER)
    extracted_files = tar.getmembers()
```